### PR TITLE
Cms transformer generator

### PR DESCRIPTION
**Pull request by cvalarida**
## Description
A generator to quickly scaffold the transformers and schemas.

Use with `yarn new:cms-content-model`.

## Testing done
Ran the thing locally a lot of times.

## Screenshots
image

The output field name defaults to the camelCase version of the input field name, but can be overwritten
image

When selecting the input schema, it shows a list of common field schemas...
image

...and we can select them using a fuzzy string match.
image

Just like the input schema, we're given a list of common output schemas.
image

It's important to note that these are just suggestions. We can type in whatever we want.
image
image

When it's all said and done, it'll tell us what files it created.
image

As an added bonus, it'll catch you if you try to create a transformer that already exists.
image

The files it creates:
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
     "lint:sass": "sass-lint --verbose",
     "mock-api": "node src/platform/testing/e2e/mockapi.js",
     "new:app": "yo @department-of-veterans-affairs/vets-website && npm run lint:js:untracked:fix",
-    "new:cms-content-model": "yo ./src/site/stages/build/process-cms-exports/generator-cms-content-model/generators/app/index.js",
+    "new:cms-content-model": "yo ./src/site/stages/build/process-cms-exports/generator/index.js && npm run lint:js:untracked:fix",
```
Comment (bkjohnson) [inline package.json:34]: The linting is a nice addition :+1:

File: src/site/stages/build/process-cms-exports/generator/templates/inputSchema [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/* eslint-disable camelcase */
+
+module.exports = {
+  type: 'object',
+  properties: {
+  <% fieldData.forEach(fd => { %>    <%- fd.inputFieldName %>: <%- fd.inputSchema %>,
+  <% }) %>
+  },
+};
```
Comment (bkjohnson) [inline src/site/stages/build/process-cms-exports/generator/templates/inputSchema:9]: This doesn't have the `required` key.  Should that be included?

File: src/site/stages/build/process-cms-exports/generator/templates/outputSchema [hunk 3]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
+  type: 'object',
+  properties: {
+  entityType: { type: 'string', enum: ['<%= entityType %>'] },
+  entityBundle: { type: 'string', enum: ['<%= entityBundle %>'] },
+  <%_ fieldData.forEach(fd => { _%>
+  <%- fd.outputFieldName %>: <%- fd.outputSchema %>,
+  <%_ }); _%>
+  },
+};
```
Comment (bkjohnson) [inline src/site/stages/build/process-cms-exports/generator/templates/outputSchema:10]: Same here regarding `required`.

File: src/site/stages/build/process-cms-exports/generator/index.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+const getFieldSchema = schemaOptions => async (answers, input = '') =>
+  fuzzy.filter(input, schemaOptions).map(el => el.original);
+
+module.exports = class extends Generator {
+  constructor(args, opts) {
+    super(args, opts);
+    this.env.adapter.promptModule.registerPrompt(
+      'autocomplete',
+      require('inquirer-autocomplete-prompt'),
```
Comment (bkjohnson) [inline src/site/stages/build/process-cms-exports/generator/index.js:34]: :sparkles:

**Review by bkjohnson — COMMENTED** [review]
Looks good, just have some comments about the schema templates not having `required`.

Also, it looks like the main difference between this generator and the previous one is that with this one we have to type out the fields we want & choose their schemas rather than select the fields from a list, is that right?

**Comment by cvalarida** [comment]
Ooh, good call with `required`. :detective: I'll add that in there.

As for the difference between this one and the last, yes, that about sums it up. Whereas the previous generator required us to copy in a JSON file from the export into the `tests/entities/` directory and add the entry to `tests/entities/index.js`, this doesn't do anything of the sort. It makes no assumptions and has no setup requirements, lowering the barrier to entry, as it were.

**Review by bkjohnson — APPROVED** [review]
:+1:

**Review by drorva — APPROVED** [review]

**Pull request merged**